pennylane/qchem/convert_openfermion.py

```diff
@@ -0,0 +1,168 @@
+
+    Args:
+        qubit_operator (QubitOperator): fermionic-to-qubit transformed operator in terms of
+            Pauli matrices
+        wires (Wires, list, tuple, dict): Custom wire mapping used to convert the qubit operator
+            to an observable terms measurable in a PennyLane ansatz.
+            For types Wires/list/tuple, each item in the iterable represents a wire label
+            corresponding to the qubit number equal to its index.
+            For type dict, only int-keyed dict (for qubit-to-wire conversion) is accepted.
+            If None, will use identity map (e.g. 0->0, 1->1, ...).
+        tol (float): tolerance value to decide whether the imaginary part of the coefficients is retained
+        return_sum (bool): flag indicating whether a ``Sum`` object is returned
+
+    Returns:
+        (pennylane.ops.Sum, pennylane.ops.LinearCombination): a linear combination of Pauli words
+
+    **Example**
+
+    >>> q_op = QubitOperator('X0', 1.2) + QubitOperator('Z1', 2.4)
+    >>> q_op
+    1.2 [X0] +
+    2.4 [Z1]
+    >>> from_openfermion(q_op)
+    1.2 * X(0) + 2.4 * Z(1)
+    """
+
+    coeffs, pl_ops = _openfermion_to_pennylane(ops, tol=tol)
+    pl_term = qml.ops.LinearCombination(coeffs, pl_ops)
+
+    if "format" in kwargs:
+        if kwargs["format"] == "Sum":
+            return qml.dot(*pl_term.terms())
+        if kwargs["format"] != "LinearCombination":
+            f = kwargs["format"]
+            raise ValueError(f"format must be a Sum or LinearCombination, got: {f}.")
+
+    return pl_term
+
+
+def to_openfermion(
+    pl_op: Union[Sum, LinearCombination, FermiWord, FermiSentence], wires=None, tol=1.0e-16
+):
+    r"""Convert a PennyLane operator to a OpenFermion ``QubitOperator`` or ``FermionOperator``.
+
+    Args:
+        pl_op (pennylane.ops.Sum, pennylane.ops.LinearCombination, pennylane.fermi.FermiWord, pennylane.fermi.FermiSentence):
+            linear combination of operators
+        wires (Wires, list, tuple, dict):
+            Custom wire mapping used to convert the qubit operator
+            to an observable terms measurable in a PennyLane ansatz.
+            For types Wires/list/tuple, each item in the iterable represents a wire label
+            corresponding to the qubit number equal to its index.
+            For type dict, only int-keyed dict (for qubit-to-wire conversion) is accepted.
+            If None, will use identity map (e.g. 0->0, 1->1, ...).
+
+    Returns:
+        (QubitOperator, FermionOperator): an OpenFermion operator
+
+    **Example**
+
+    >>> pl_term = 1.2 * qml.X(0) + 2.4 * qml.Z(1)
+    >>> pl_term
+    1.2 * X(0) + 2.4 * Z(1)
+    >>> q_op = to_openfermion(q_op)
+    >>> q_op
+    1.2 [X0] +
+    2.4 [Z1]
+    """
+    return _to_openfermion_dispatch(pl_op, wires=wires, tol=tol)
+
+
+@singledispatch
+def _to_openfermion_dispatch(pl_op, wires=None, tol=1.0e-16):
```

Comment:
It should be `tol=tol`?

Reply:
Why? In any dispatch function, `tol` is a default argument. Using `tol=tol` results in an error:
```
NameError: name 'tol' is not defined
```

Reply:
I could not get rid of the imaginary parts in an example, thought maybe the `tol` is not propagated properly.

Reply:
I have adjusted the example accordingly. The underlying `_pennylane_to_openfermion` method makes the coefficients of the `complex` type. Even if the tolerance is not exceeded, the imaginary part of the coefficients will not be discarded but will be set to zero.